consumer/shard_api.go

```diff
@@ -128,6 +128,11 @@ func ShardApply(ctx context.Context, srv *Service, req *pc.ApplyRequest) (*pc.Ap
 		}
 	}
 
+	if len(ops) == 0 {
```

Comment:
I'm missing why this would be required given the check of `len(ops)` below. The current behavior is to go ahead and do a no-op etcd transaction, but to only wait on the revision if there was at least one applied operation.

I think you meant this to be part of `ShardUnassign`, below? For consistency I'd recommend having the same handling in `ShardUnasign` as is currently done with `ShardApply` (do the no-op transaction, but only wait if empty).

Reply:
Doh. I don't know how I ended up modifying `ShardApply`, but that wasn't the intention. :facepalm: